lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -2135,26 +2135,57 @@ namespace Js
     {
         charcount_t count = pThis->GetLength();
 
-        BufferStringBuilder builder(count, pThis->type->GetScriptContext());
-        const wchar_t *inStr = pThis->GetString();
-        wchar_t *outStr = builder.DangerousGetWritableBuffer();
+        const wchar_t* inStr = pThis->GetString();
+        const wchar_t* i = inStr; 
+        const wchar_t* iLim = i + count;
+        
+        // first, try to see if the string needs recasing at all
+        while(i < iLim)
+        {
+            
+            // copy the current char into a temporary variable, then case it
+            wchar_t inChar = *i;
+            wchar_t outChar = *i;
+            CharUpperBuffW(&outChar, 1);
```

Comment:
I don't think this is the right way to check for this. This method `ToCaseCore` can be used for converting to Upper- or Lower-case.

Additionally I don't think this method will be inlined, so there's a lot of function call overhead added here. What are the gains for pre-processing the string like this when you're attempting to run the conversion on every character anyway?

There might be something in the general idea of this pull request but I doubt this implementation will result in a performance improvement.